### DISCUSSION TITLE
Inject model in ->hidden() on an action group

### DISCUSSION THREAD

**Discussion by sinnbeck** [filamentphp/filament#7663] [Help] [answered]
This is using filament v3.

I am trying to inject my model into the hidden() method on an action group, but it does not seem to resolve the model instance. The form fields works just as expected

```php
return $table
    ->query(Application::query())
    ->action([
        ActionGroup::make([
            ActionGroup::make([TextInput('name')])
                ->hidden(function (Application $application) {
                    return $application->status === 'rejected'; //this is not resolved. Its just an empty model
                })
                ->icon('heroicon-o-x-circle')
                ->grouped()
                ->color('danger'),

        ]),
    ]);
```
Does anyone know how to fix this ?

**Comment by danharrin** [comment, marked as answer]
Can you try replacing `Application $application` with `ActionGroup $group`, and then using `$group->getRecord()`? I don't think we're automatically passing in record to closures yet for action groups

**Reply by sinnbeck** [reply]
Thanks! That seems to work! I just assumed it was possible the other way as it works on Actions :)

I will just do this instead

**Reply by danharrin** [reply]
`$record` should work, I'll add it to my todo list

**Reply by danharrin** [reply]
https://github.com/filamentphp/filament/commit/2205b53dc7c7ab7653a128229d2baafd05a098e0

**Reply by mahjoubkzbr** [reply]
hey there, is there a way to hide action groups of the table in a relation manager? my approach works fine on the resource table, but on the relation manager i get an exception `Attempt to read property "is_active" on null` when calling the resource table.

```
Tables\Actions\ActionGroup::make([
                    Tables\Actions\EditAction::make(),
                    Tables\Actions\ViewAction::make(),
                    Tables\Actions\Action::make('Archivieren')
                        ->icon('far-folder-closed')
                        ->hidden(fn(Mandate $record) => !$record->is_active)
                        ->action(function (Mandate $record) {
                            $record->archive();
                            Notification::make()
                                ->title('Der Auftrag wurde archiviert.')
                                ->success()
                                ->send();
                        }),
                    Tables\Actions\DeleteAction::make(),
                    Tables\Actions\RestoreAction::make(),
                ])->hidden(fn($record) => !$record->is_active),
```
